src/cpp/src/utils/paged_attention_transformations.cpp

```diff
@@ -50,21 +49,30 @@ void set_kv_cache_type_and_shape(std::shared_ptr<ov::Model> model, DeviceConfig&
     for (size_t idx = 0; idx < num_decoder_layers; idx++) {
         KVHeadConfig& config = kv_heads_config[idx];
 
-        auto key_shape = key_cache_params[std::string("key_cache.") + std::to_string(idx)]->get_partial_shape();
+        auto k = key_cache_params[std::string("key_cache.") + std::to_string(idx)];
+        auto key_shape = k->get_partial_shape();
         config.num_k_heads = key_shape[1].get_length();
         config.k_head_size = key_shape[2].get_length();
 
-        auto value_shape = value_cache_params[std::string("value_cache.") + std::to_string(idx)]->get_partial_shape();
+        auto v = value_cache_params[std::string("value_cache.") + std::to_string(idx)];
+        auto value_shape = v->get_partial_shape();
         config.num_v_heads = value_shape[1].get_length();
         config.v_head_size = value_shape[2].get_length();
     }
+
+    // save information about KV caches in device_config
+    // and create device dependent KV cache shapes
     device_config.set_kv_head_configs(kv_heads_config);
 
     for (size_t idx = 0; idx < num_decoder_layers; idx++) {
         auto k = key_cache_params[std::string("key_cache.") + std::to_string(idx)];
         auto v = value_cache_params[std::string("value_cache.") + std::to_string(idx)];
-        k->set_element_type(device_config.get_cache_precision());
-        v->set_element_type(device_config.get_cache_precision());
+
+        // allow a plugin to automatically set KV cache precisions
+        k->set_element_type(ov::element::undefined);
+        v->set_element_type(ov::element::undefined);
```

Comment:
@luo-cheng2021 @sshlyapn 
Here I make KV cache precisions in original PA model to be `undefined` and CB wants to rely on the same logic for KV cache precisions as plugins make for SPDA case.

The rest of other changes serve to respect precisions for `ov::InferRequest` of produced by plugins

Could you please make branches for CPU / GPU in OV repo to comply with these changes?